File: src/test/java/com/leetcode/top/interview/easy/LongestCommonPrefixTest.java
package com.leetcode.top.interview.easy;

import org.junit.Assert;
import org.junit.Test;

/**
 * @author wezhyn
 * @since 08.23.2020
 */
public class LongestCommonPrefixTest {
    @Test
    public void test() {
        Assert.assertEquals("a", new LongestCommonPrefix().longestCommonPrefix(new String[]{"a", "ab"}));
        Assert.assertEquals("fl", new LongestCommonPrefix().longestCommonPrefix(new String[]{"flower", "flow", "flight"}));
        Assert.assertEquals("", new LongestCommonPrefix().longestCommonPrefix(new String[]{"dog", "racecar", "car"}));
    }

}

File: leetcode-kotlin/src/test/kotlin/com/wezhyn/learn/string/LongestCommonPrefixTest.kt
package com.wezhyn.learn.string

import org.junit.Assert.assertEquals
import org.junit.Test

/**
 *
 * @author wezhyn
 * @since 08.23.2020
 */
class LongestCommonPrefixTest {

    @Test
    fun longestCommonPrefix() {
        assertEquals("a", LongestCommonPrefix().longestCommonPrefix(arrayOf("a", "ab")))
        assertEquals("", LongestCommonPrefix().longestCommonPrefix(arrayOf("dog", "racecar", "car")))
        assertEquals("fl", LongestCommonPrefix().longestCommonPrefix(arrayOf("flower", "flow", "flight")))
    }
}